saw-core/src/Verifier/SAW/Simulator/Prims.hs

```diff
@@ -864,42 +842,47 @@ shiftOp bp vecOp wordIntOp wordOp =
             zb <- toBool <$> force z
             VWord <$> wordIntOp zb xw (toInteger (min i n))
           _ -> panic $ "shiftOp: " ++ show xs
-      VToNat (VVector iv) -> do
+      VBVToNat _sz (VVector iv) -> do
         bs <- V.toList <$> traverse (fmap toBool . force) iv
         case xs of
-          VVector xv -> VVector <$> shifter muxVector (\v i -> return (vecOp z v i)) xv bs
+          VVector xv -> VVector <$> shifter (muxVector n tp) (\v i -> return (vecOp z v i)) xv bs
           VWord xw -> do
             zb <- toBool <$> force z
             VWord <$> shifter (bpMuxWord bp) (wordIntOp zb) xw bs
           _ -> panic $ "shiftOp: " ++ show xs
-      VToNat (VWord iw) ->
+      VBVToNat _sz (VWord iw) ->
         case xs of
           VVector xv -> do
             bs <- V.toList <$> bpUnpack bp iw
-            VVector <$> shifter muxVector (\v i -> return (vecOp z v i)) xv bs
+            VVector <$> shifter (muxVector n tp) (\v i -> return (vecOp z v i)) xv bs
           VWord xw -> do
             zb <- toBool <$> force z
             VWord <$> wordOp zb xw iw
           _ -> panic $ "shiftOp: " ++ show xs
+
+      VIntToNat _i -> error "shiftOp: symbolic integer TODO"
```

Comment:
This should be implementable; we will need to produce a big chain of nested if-then-else expressions with index equality tests.